File: src/services/besc_host.js
class BESC_ESS_Host{

    constructor(host){
        this.host = host;
    }

    static createDefault(){
        return new BESC_ESS_Host("http://carboapi.besc.online/besc-data");
    }

    setEndPoint(endpoint){
        this.endpoint = endpoint;
    }

    setProjectId(project_id){
        this.project_id = project_id;
    }

    buildUrl(){

        var endpoint = this.endpoint.replace(":project_id", this.project_id);

        return this.host + basePath + endpoint;
    }

}

const endpoints = {
    debug: "/debugs/:project_id",
    project: "/project/:project_id"
};

const basePath = "/energysaving";

module.exports.default = BESC_ESS_Host;
module.exports.endpoints = endpoints;

File: src/services/besc_host.py
class BESC_ESS_Host:

    def __init__(self, host):
        self.host = host

    '''
    def createDefault(self):
        return BESC_ESS_Host("http://carboapi.besc.online/besc-data");
    '''

    def setEndPoint(self, endpoint):
        self.endpoint = endpoint

    def setProjectId(self, project_id):
        self.project_id = project_id

    def buildUrl(self):
        endpoint = str(self.endpoint).replace(":project_id", self.project_id)
        return self.host + basePath + endpoint

endpoints = {
    "debug": "/debugs/:project_id",
    "project": "/project/:project_id"
}

basePath = "/energysaving"
